File: src/Gopher/ClientApp/src/app/modules/project/models/project.ts
import { ProjectTask } from "../../projecttask/models/projecttask";

export interface Project{
    id:string,
    userID:string,
    title: string,
    description: string,
    ProjectTasks: Array<ProjectTask>,
    //projecttasks: Array<ProjectTask>,
    date:Date
}

File: src/Gopher/Models/Project.cs
namespace Gopher.Models
{
    public class Project
    {
        public int ID { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public string UserID { get; set; }

        //navigation properties
        public virtual ICollection<ProjectTask> ProjectTasks { get; set; } = new List<ProjectTask>();

        public Project() { }

        public Project(IEnumerable<ProjectTask> projectTasks)
        {
            ProjectTasks = projectTasks.ToList();
        }
    }
}